sdk/Lusid.Drive.Sdk/Extensions/FilesApiExtensions.cs

```diff
@@ -0,0 +1,263 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Lusid.Drive.Sdk.Api;
+using Lusid.Drive.Sdk.Client;
+using Lusid.Drive.Sdk.Model;
+using RestSharp.Portable;
+using RestSharp.Portable.HttpClient.Impl.Http;
+
+namespace Lusid.Drive.Sdk.Extensions
+{
+    /// <summary>
+    /// Represents a collection of Extension Methods to interact with Rest Sharp using a Stream instead of a byte array
+    /// </summary>
+    public static class FilesApiExtensions
+    {
+        /// <summary>
+        ///     Uploads data to a file as a stream instead of as a byte array.
+        ///     The function is asynchronous as RestSharp only offers asynchronous methods.
+        ///     This method was created in order to support file uploads as Streams as this is less taxing on memory
+        ///     than byte arrays. This would be better for large file uploads or in the case where multiple users are
+        ///     uploading at the same time, causing a lot of memory to be consumed.
+        /// </summary>
+        /// <param name="xLusidDriveFilename">File name.</param>
+        /// <param name="xLusidDrivePath">File path.</param>
+        /// <param name="contentLength">The size in bytes of the file to be uploaded</param>
+        /// <param name="body">The data to upload</param>
+        /// <returns>Storage Object as a Task</returns>
+        public static async Task<StorageObject> UploadAsStreamAsync(this IFilesApi api,
+            string xLusidDriveFilename,
+            string xLusidDrivePath,
+            int? contentLength,
+            Stream body)
+        {
+            // Verify that 'xLusidDriveFilename' is set
+            if (xLusidDriveFilename == null)
+            {
+                throw new ApiException(400, 
+                    "Missing required parameter 'xLusidDriveFilename' when calling ExtensionsMethods->CreateFile");
+            }
+
+            // Verify that 'xLusidDrivePath' is set
+            if (xLusidDrivePath == null)
+            {
+                throw new ApiException(400,
+                    "Missing required parameter 'xLusidDrivePath' when calling ExtensionsMethods->CreateFile");
+            }
+
+            // Verify that 'contentLength' is set
+            if (contentLength == null)
+            {
+                throw new ApiException(400, 
+                    "Missing required parameter 'contentLength' when calling ExtensionsMethods->CreateFile");
+            }
+
+            // Verify that 'body' is set
+            if (body == null)
```

Comment:
do you need to check for an empty body or is that expected to create an empty file?

Reply:
Hmm I guess I'm not sure. I was following the style in which some of the other methods had been written, in this case the CreateFile method: https://github.com/finbourne/drive-sdk-csharp-preview/blob/master/sdk/Lusid.Drive.Sdk/Api/FilesApi.cs#L452

Reply:
Got rid of the check